skymap_scanner/server/start_scan.py

```diff
@@ -162,31 +186,31 @@ def refine_vertex_time(vertex, time, direction, pulses, omgeo):
             return time
         return min_t + adj_d/ccc
 
-    def generate_pframes(self) -> Iterator[icetray.I3Frame]:
-        """Yield PFrames to be reco'd."""
+    def gen_new_pixel_pframes(
+        self,
+        already_sent_pixels: Set[SentPixel],
+    ) -> Iterator[icetray.I3Frame]:
+        """Yield pixels (PFrames) to be reco'd."""
+
+        def pixel_already_sent(pixel: Tuple[icetray.I3Int, icetray.I3Int]) -> bool:
+            # Has this pixel already been sent? Ignore the position-variation id
+            for sent_pixel in already_sent_pixels:
+                if sent_pixel.nside == pixel[0] and sent_pixel.pixel_id == pixel[1]:
```

Comment:
How would it handle a scenario where we have N>1 position variations, and each pixel has finished reconstructing only one positional variation each so that it crosses the threshold for refinment, but no best-pixelreco has been recorded since none of the pixels are complete?

Reply:
the threshold for refinement is calculated by # of finished whole pixels (that's what gets stored in `nsides_dict`). However I could add an additional filter check at the `yield` in `gen_new_pixel_pframes()`. What do you think?

Reply:
I think it makes sense now that it treats it using the # finished pixels. What do you mean by additional filter check to look at the variation id?

Reply:
I'm not sure, I thnk I was hungry :sweat_smile: Sounds like we're good on this one